extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -166,4 +166,62 @@ public synchronized void findKeys(@NotNull final Consumer<IcebergTableLocationKe
                     String.format("%s:%d - error finding Iceberg locations", tableAdapter, snapshot.snapshotId()), e);
         }
     }
+
+    /**
+     * Update the snapshot to the latest snapshot from the catalog if
+     */
+    protected boolean maybeUpdateSnapshot() {
```

Comment:
Should we assert that we hold a lock? Or just document that this should be `synchronized`? Or just make it `synchronized`? We could do compareAndSet, but it's overkill since this is always called from synchonized code.